src/main/java/org/icgc/argo/workflow_management/util/VolumeMounts.java

```diff
@@ -0,0 +1,121 @@
+   */
+  public static List<String> extract(
+      NextflowProperties.K8sProperties k8sProperties, WorkflowEngineParams workflowEngineParams) {
+    return Optional.ofNullable(k8sProperties.getVolMounts())
+        .map(
+            volMounts ->
+                extract(
+                    volMounts,
+                    Stream.of(
+                            workflowEngineParams.getLaunchDir(),
+                            workflowEngineParams.getProjectDir(),
+                            workflowEngineParams.getWorkDir())
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList())))
+        .orElseThrow(
+            () ->
+                new IllegalStateException(
+                    "At least one volMount must be configured in order for Nextflow to run in Kubernetes"));
+  }
+
+  /**
+   * Given a list of volMounts and paths, returns the volMounts required for the paths provided
+   * where a volMount (with the format claim-name:mount-path) is considered required if the path
+   * starts with the volume mount path. If no matches are found and there is at least one volMount,
+   * the first volMount will be returned as the single item in a list as at least one volMount is
+   * required
+   *
+   * @param volMounts - list of volMounts to search through
+   * @param paths - list of paths to use as the search subject
+   * @return the matching volMounts given the search
+   */
+  public static List<String> extract(@NonNull List<String> volMounts, @NonNull List<String> paths) {
+    return volMounts.stream()
+        .filter(volMount -> paths.stream().anyMatch(getPathMatchFunctionForVolMount(volMount)))
+        // volMounts need to be provided as a list,
+        // however we only care about distinct values
+        .distinct()
+        .collect(
+            Collectors.collectingAndThen(
+                Collectors.toList(), getDefaultVolMountIfEmptyFunctionForVolMount(volMounts)));
+  }
+
+  /**
+   * Utility overload method that calls extract(@NonNull List<String> volMounts, @NonNull
+   * List<String> paths)
+   *
+   * @param volMounts - list of volMounts to search through
+   * @param path - the path to use as the search subject
+   * @return the matching volMounts given the search
+   */
+  public static List<String> extract(@NonNull List<String> volMounts, @NonNull String path) {
+    return extract(volMounts, List.of(path));
+  }
+
+  /**
+   * Generates a function for matching path to volume mount
+   *
+   * @param volMount a single volMount entry from the list in application properties
+   * @return a function which evaluates that the path provided matches the pattern for the volMount
+   *     used to generate this function
+   */
+  private static Predicate<String> getPathMatchFunctionForVolMount(String volMount) {
+    return path -> getPatternForVolMount(volMount).matcher(path).find();
```

Comment:
why not path.startsWith(volMount.split(":")[1])?

Reply:
yup, I totally forgot about the built in string methods :sweat::palm_tree: